l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1183,6 +1183,21 @@ def _prepare_invoice_values(self, fatt, fatturapa_attachment, FatturaBody, partn
         self.set_e_invoice_lines(FatturaBody, invoice_data)
         return invoice_data
 
+    def _new_write_vals(self, invoice):
+        """
+        Update changing values only
+        """
+        to_return = invoice._convert_to_write(invoice._cache)
+        to_remove = set()
+        for k, val in to_return.items():
+            if (
+                k.endswith("_ids") and val[0][0] == 6 and val[0][2] == invoice[k].ids
```

Comment:
**issue**: checking if the field name ends with `_ids` is very unreliable. If you want to check if you have a m2m or o2m, you can do so by accessing `invoice._fields`. The field object has a `type`

An example in the screenshot below

![image](https://github.com/OCA/l10n-italy/assets/78726989/97bee7eb-0c7b-46e8-bea7-64a99e677595)

Reply:
I'm unsure if there are additional edge cases that we're not considering here, because `_convert_to_write` could container a tuple of type `(6, 0, ids)` but also a `(5,...)` or a `(0, ...)`

https://github.com/OCA/OCB/blob/14.0/odoo/fields.py#L2988-L3025

I wonder if we could think about a more reliable way of checking. Maybe using the _cache object before applying `_convert_to_write` to it?